test/e2e/kustomizations_test.go

```diff
@@ -119,3 +123,28 @@ func patchLivenessProbes(baseManifestReader io.Reader, deployment k8stypes.Names
 	}
 	return kubectl.GetKustomizedManifest(kustomization, baseManifestReader)
 }
+
+// patchReadinessProbePath patches the given deployment's path of readiness probe.
+// Kong gateway supports endpoint `/status/ready` since 3.4, and prior versions uses `/status`,
+// so we need to change the path of readiness probe of Kong gateway deployment when its version is < 3.4.
+func patchReadinessProbePath(baseManifestReader io.Reader, deployment k8stypes.NamespacedName, probePath string) (io.Reader, error) {
+	kustomization := types.Kustomization{
+		Patches: []types.Patch{
+			{
+				Patch: fmt.Sprintf(readinessProbePathPatch, probePath),
+				Target: &types.Selector{
+					ResId: resid.ResId{
+						Gvk: resid.Gvk{
+							Group:   "apps",
+							Version: "v1",
+							Kind:    "Deployment",
+						},
+						Name:      deployment.Name,
+						Namespace: deployment.Namespace,
+					},
+				},
+			},
+		},
+	}
```

Comment:
Sorry I wasn't clear on this. I meant to add a function like:

```
func formatKustomizePatchFirstContainerHTTPReadinessProbe(probePath string) string {
	const readinessProbePathPatch = `` +
		`- op: replace` +
		`  path: /spec/template/spec/containers/0/readinessProbe/httpGet/path` +
		`  value: %s`
	return fmt.Sprintf(readinessProbePathPatch, probePath)
}
```

and then

```suggestion
	kustomization := types.Kustomization{
		Patches: []types.Patch{
			{
				Patch: formatKustomizePatchFirstContainerHTTPReadinessProbe(probePath),
				Target: &types.Selector{
					ResId: resid.ResId{
						Gvk: resid.Gvk{
							Group:   "apps",
							Version: "v1",
							Kind:    "Deployment",
						},
						Name:      deployment.Name,
						Namespace: deployment.Namespace,
					},
				},
			},
		},
	}
```

this way we can encapsulate the patch formatting so that there are no mistakes with e.g. using `fmt.Sprintf` without format param.

Reply:
done.